### PR TITLE
Split `https_availability` audit.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Allow more fine-grained control for excluding audits. For example, https://github.com/Homebrew/homebrew-cask/pull/163887 shouldn't use `ci-skip-livecheck` to skip the homepage audit when there isn't even a livecheck anymore.

### CODE REVIEW

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -770,32 +770,48 @@ def audit_reverse_migration
...
 
-      validate_url_for_https_availability(cask.homepage, SharedAudits::URL_TYPE_HOMEPAGE, cask.token, cask.tap,
-                                          user_agents:   user_agents,
-                                          check_content: true,
-                                          strict:        strict?)
+      validate_url_for_https_availability(
+        homepage, SharedAudits::URL_TYPE_HOMEPAGE,
+        user_agents:   user_agents,
+        check_content: true,
+        strict:        strict?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:787]: Is there actually a reason to only make the `homepage` audit strict instead of all of them?

Comment (reitermarkus) [reply Library/Homebrew/cask/audit.rb:787]: Ah, it's passed to `curl_check_http_content` for additional checks and not actually used for audit severity.

**Review by apainintheneck — APPROVED** [review]
Makes sense

**Comment by bevanjkay** [comment]
Having the extra label makes sense 👍🏻

Do we need to consider some adjustments with the way deprecated/disabled casks are handled in general for auditing? Probably an AGM discussion though.

**Comment by reitermarkus** [comment]
> the way deprecated/disabled casks are handled

I think we already have some handling that makes sense, e.g. ignoring livecheck for them. In this specific case with the `knotes` cask it's not even HTTPS availability that's the problem, but availability in general, since both HTTP and HTTPS return `503 Service Unavailable`. Still worth discussing though.

**Comment by reitermarkus** [comment]
Merge together with https://github.com/Homebrew/homebrew-cask/pull/163916.

**Comment by bevanjkay** [comment]
@reitermarkus I'm seeing a CI failure related to the change here - https://github.com/Homebrew/homebrew-cask/actions/runs/7440382751/job/20241842575?pr=163945#step:11:63

**Pull request merged**